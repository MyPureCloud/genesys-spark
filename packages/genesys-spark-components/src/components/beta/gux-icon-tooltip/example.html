<h1>gux-icon-tooltip-beta</h1>

<h2>Default</h2>
<gux-icon-tooltip-beta icon-name="fa/bell-regular">
  <span slot="content">This is some content text</span>
</gux-icon-tooltip-beta>
