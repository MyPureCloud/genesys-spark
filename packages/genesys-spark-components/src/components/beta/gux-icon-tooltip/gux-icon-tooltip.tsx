import { Component, Element, h, JSX, Prop, Method } from '@stencil/core';
import { trackComponent } from '@utils/tracking/usage';
import { Placement } from '@floating-ui/dom';

import { GuxIconIconName } from '../../stable/gux-icon/gux-icon.types';
import { getSlotTextContent } from '../../../utils/dom/get-slot-text-content';

/**
 * @slot content - Required slot for tooltip and screenreader content
 */

@Component({
  styleUrl: 'gux-icon-tooltip.scss',
  tag: 'gux-icon-tooltip-beta',
  shadow: true
})
export class GuxIconTooltip {
  private tooltip: HTMLGuxTooltipBetaElement;

  @Element()
  private root: HTMLElement;

  @Prop()
  iconName: GuxIconIconName;

  @Prop({ mutable: true })
  placement: Placement = 'right';

  componentWillLoad(): void {
    trackComponent(this.root);
  }

  /*
   * Show tooltip
   */
  @Method()
  async showTooltip(): Promise<void> {
    return await this.tooltip.showTooltip();
  }

  /*
   * Hide tooltip
   */
  @Method()
  async hideTooltip(): Promise<void> {
    return await this.tooltip.hideTooltip();
  }

  render(): JSX.Element {
    return (
      <div class="gux-icon-tooltip">
        <gux-screen-reader-beta>
          <slot name="content"></slot>
        </gux-screen-reader-beta>
        <gux-icon icon-name={this.iconName} size="small" decorative></gux-icon>
        <gux-tooltip-beta
          placement={this.placement}
          ref={el => (this.tooltip = el)}
        >
          <div slot="content">{getSlotTextContent(this.root, 'content')}</div>
        </gux-tooltip-beta>
      </div>
    ) as JSX.Element;
  }
}
