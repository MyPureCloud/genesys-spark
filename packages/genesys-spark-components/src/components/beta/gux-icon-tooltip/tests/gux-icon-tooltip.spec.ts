import { newSpecPage } from '@test/specTestUtils';
import { GuxIconTooltip } from '../gux-icon-tooltip';

const components = [GuxIconTooltip];
const language = 'en';

describe('gux-icon-tooltip-beta', () => {
  describe('#render', () => {
    [
      '<gux-icon-tooltip-beta></gux-icon-tooltip-beta>',
      '<gux-icon-tooltip-beta><span slot="content">This is some tooltip text</span></gux-icon-tooltip-beta>',
      '<gux-icon-tooltip-beta icon-name="fa/bell-regular"></gux-icon-tooltip-beta>',
      '<gux-icon-tooltip-beta icon-name="fa/bell-regular"><span slot="content">This is some tooltip text</span></gux-icon-tooltip-beta>'
    ].forEach((html, index) => {
      it(`should render component as expected (${index + 1})`, async () => {
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });
  });
});
