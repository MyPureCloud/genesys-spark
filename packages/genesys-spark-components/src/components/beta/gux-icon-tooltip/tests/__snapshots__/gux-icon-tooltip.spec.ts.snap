// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-icon-tooltip-beta #render should render component as expected (1) 1`] = `
<gux-icon-tooltip-beta>
  <mock:shadow-root>
    <div class="gux-icon-tooltip">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content"></div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
</gux-icon-tooltip-beta>
`;

exports[`gux-icon-tooltip-beta #render should render component as expected (2) 1`] = `
<gux-icon-tooltip-beta>
  <mock:shadow-root>
    <div class="gux-icon-tooltip">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content">
          This is some tooltip text
        </div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
  <span slot="content">
    This is some tooltip text
  </span>
</gux-icon-tooltip-beta>
`;

exports[`gux-icon-tooltip-beta #render should render component as expected (3) 1`] = `
<gux-icon-tooltip-beta icon-name="fa/bell-regular">
  <mock:shadow-root>
    <div class="gux-icon-tooltip">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" icon-name="fa/bell-regular" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content"></div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
</gux-icon-tooltip-beta>
`;

exports[`gux-icon-tooltip-beta #render should render component as expected (4) 1`] = `
<gux-icon-tooltip-beta icon-name="fa/bell-regular">
  <mock:shadow-root>
    <div class="gux-icon-tooltip">
      <gux-screen-reader-beta>
        <slot name="content"></slot>
      </gux-screen-reader-beta>
      <gux-icon decorative="" icon-name="fa/bell-regular" size="small"></gux-icon>
      <gux-tooltip-beta placement="right">
        <div slot="content">
          This is some tooltip text
        </div>
      </gux-tooltip-beta>
    </div>
  </mock:shadow-root>
  <span slot="content">
    This is some tooltip text
  </span>
</gux-icon-tooltip-beta>
`;
